cores/esp8266/core_esp8266_waveform.c

```diff
@@ -38,84 +38,41 @@
 */
 
 #include <Arduino.h>
+#include "ets_sys.h"
 #include "core_esp8266_waveform.h"
 
-// Need speed, not size, here
-#pragma GCC optimize ("O2")
-
 // Maximum delay between IRQs
 #define MAXIRQUS (10000)
 
-// If the cycles from now to an event are below this value, perform it anyway since IRQs take longer than this
-#define CYCLES_FLUFF (100)
-
-// Macro to get count of predefined array elements
-#define countof(a) ((size_t)(sizeof(a)/sizeof(a[0])))
-
-// Set/clear *any* GPIO
-#define SetGPIOPin(a) do { if (a < 16) { GPOS |= (1<<a); } else { GP16O |= 1; } } while (0)
-#define ClearGPIOPin(a) do { if (a < 16) { GPOC |= (1<<a); } else { GP16O &= ~1; } } while (0)
-// Set/clear GPIO 0-15
+// Set/clear GPIO 0-15 by bitmask
 #define SetGPIO(a) do { GPOS = a; } while (0)
 #define ClearGPIO(a) do { GPOC = a; } while (0)
 
+// Convert from us to ESP8266 clocks
+#define MicrosecondsToCycles(m) (clockCyclesPerMicrosecond() * (m))
```

Comment:
This is already defined in Arduino.h as  `clockCyclesPerMicrosecond`

Reply:
Yes, ` clockCyclesPerMicrosecond` is defined, but I need it for a fixed # of milliseconds (m) bit.

Reply:
I don't follow, the Arduino.h macro looks the same as this one:
```
#define microsecondsToClockCycles(a) ( (a) * clockCyclesPerMicrosecond() )
```

Also: milliseconds? I'm confused now.

Reply:
Ah, darn it.  I did mean *micro*seconds and I didn't see that macro for some reason.  I'll replace the lot out...

Reply:
Done in latest push.